google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -1327,4 +1329,54 @@ public void floatIncrement() throws ExecutionException, InterruptedException {
       this.error = error;
     }
   }
+
+  @Test
+  public void getAllWithObserver() throws Exception {
+    DocumentReference ref = randomColl.document("doc1");
+    ref.set(ALL_SUPPORTED_TYPES_MAP).get();
+
+    DocumentReference ref1 = randomColl.document("doc2");
+    ref1.set(ALL_SUPPORTED_TYPES_MAP).get();
+
+    DocumentReference ref2 = randomColl.document("doc3");
+    ref2.set(ALL_SUPPORTED_TYPES_MAP).get();
+    ref2.delete();
```

Comment:
You don't need to set and delete a document in order for it to show up as "missing". You can just query a non-existing document that you never created.

Reply:
done